lib/hiera/backend/vault_backend.rb

```diff
@@ -61,13 +82,21 @@ def lookup_generic(key, scope)
           return nil if secret.nil?
 
           Hiera.debug("[hiera-vault] Read secret: #{key}")
-          if @config[:default_field]
+          if @config[:default_field] and secret.data.has_key?(@config[:default_field].to_sym) and secret.data.length == 1
```

Comment:
Why is the `length == `1 check necessary?

e.g. if `:default_field = "value"` and you have `vault write secret/key value=x other=y else=z`, then the `other` and `else` fields should just be ignored rather.